addons/api/addon/serializers/credential-library.js

```diff
@@ -14,16 +17,39 @@ export default class CredentialLibrarySerializer extends ApplicationSerializer {
    * @method serialize
    * @param {Snapshot} snapshot
    */
-  serialize() {
+  serialize(snapshot) {
+    switch (snapshot.record.type) {
+      case TYPE_CREDENTIAL_LIBRARY_VAULT_GENERIC:
+        return this.serializeVaultGeneric(...arguments);
+      case TYPE_CREDENTIAL_LIBRARY_VAULT_SSH_CERTIFICATE:
+      default:
+        return super.serialize(...arguments);
```

Comment:
Once we introduce formal resource types, an attempt to serialise an unknown type is probably an error.

Reply:
🤔 I kept it like this since the default (which shouldn't be hit) would just do the least amount of extra serialization. I actually feel like maybe if we get to the default case we should be erroring out since that would be unexpected